SU2_CFD/include/solver_structure.hpp

```diff
@@ -10093,6 +10106,382 @@ class CTransLMSolver: public CTurbSolver {
   su2double *rhsReth;    /*!< \brief right hand side of implicit linear system. */
 };
 
+/*!
+ * \class CScalarSolver
+ * \brief Main class for defining the solver for scalar transport eqns.
+ * \ingroup Scalar_Model
+ * \author T. Economon
+ */
+class CScalarSolver : public CSolver {
+protected:
+  su2double *FlowPrimVar_i,      /*!< \brief Store the flow solution at point i. */
+  *FlowPrimVar_j,                /*!< \brief Store the flow solution at point j. */
+  *lowerlimit,                   /*!< \brief contains lower limits for turbulence variables. */
+  *upperlimit;                   /*!< \brief contains upper limits for turbulence variables. */
+  su2double*** Inlet_ScalarVars; /*!< \brief Scalar variables at inlet profiles */
+  unsigned long nMarker,         /*!< \brief Total number of markers using the grid information. */
+  *nVertex;                      /*!< \brief Store nVertex at each marker for deallocation */
+  unsigned short Inlet_Position; /*!< \brief Column index for scalar variables in inlet files. */
+  su2double *Scalar_Inf;         /*!< \brief Array of far-field values for the scalar variables. */
+
+  /* Sliding mesh variables */
+  
+  su2double ****SlidingState;
+  int **SlidingStateNodes;
+  
+public:
+  
+  /*!
+   * \brief Constructor of the class.
+   */
+  CScalarSolver(void);
+  
+  /*!
+   * \brief Destructor of the class.
+   */
+  virtual ~CScalarSolver(void);
+  
+  /*!
+   * \brief Constructor of the class.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] config - Definition of the particular problem.
+   */
+  CScalarSolver(CGeometry* geometry, CConfig *config);
+  
+  /*!
+   * \brief Compute the spatial integration using a upwind scheme.
+   * \param[in] geometry - Geometrical definition of the problem.
+   * \param[in] solver_container - Container vector with all the solutions.
+   * \param[in] numerics - Description of the numerical method.
+   * \param[in] config - Definition of the particular problem.
+   * \param[in] iMesh - Index of the mesh in multigrid computations.
+   */
+  
+  void Upwind_Residual(CGeometry *geometry,
+                       CSolver **solver_container,
```

Comment:
Are we supposed to shorten the names? Otherwise we start having inconsistencies between driver, solver, etc.